src/Tests.IntegrationTests/Helpers/RequestHelper.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Net;
+using System.Text;
+using System.Text.Json;
+using Tests.IntegrationTests.Models;
+
+namespace Tests.IntegrationTests.Helpers
+{
+    public static class RequestHelper
+    {
+        /// <summary>
+        /// Simple helper to handle http requests and deserialisation of result
+        /// </summary>
+        /// <typeparam name="T">Type that the result shoudl be deserialised to</typeparam>
+        /// <param name="httpClient">A HttpClient to use/reuse</param>
+        /// <param name="verb">Which Http verb to use</param>
+        /// <param name="uri">The Uri to send the request to</param>
+        /// <param name="uri">The Uri to send the request to</param>
+        /// <param name="apiKey">The value to send as x-api-key header</param>
+        /// <returns>Object representing deserialised result of type defined by T</returns>
+        /// <exception cref="NotImplementedException">If verb is not in expected range.</exception>
+        public static async Task<T> Request<T>(HttpClient httpClient, Verb verb, Uri uri, string apikey, string body = "")
+        {
+            if (httpClient == null)
+            {
+                throw new ArgumentNullException(nameof(httpClient));
+            }
+
+            // Add api key header unless it already exists
+            if (!httpClient.DefaultRequestHeaders.Contains("x-api-key"))
+            {
+                httpClient.DefaultRequestHeaders.Add("x-api-key", apikey);
+            }
```

Comment:
Since you're creating a `HttpClient` in each test, you can simplify the code by moving this to wherever you are creating the client. It will save you having to pass it every time you call.

That might also make this method redundant, reducing your code complexity further.

Reply:
Good suggestion, I've refactored a little to create the client with the header in the constructors. Thanks